src/test/java/org/prebid/server/settings/S3ApplicationSettingsTest.java

```diff
@@ -0,0 +1,404 @@
+
+        given(s3AsyncClient.getObject(
+                eq(GetObjectRequest.builder()
+                        .bucket(BUCKET)
+                        .key("%s/%s.json".formatted(ACCOUNTS_DIR, "accountId"))
+                        .build()),
+                any(AsyncResponseTransformer.class)))
+                .willReturn(CompletableFuture.completedFuture(
+                        ResponseBytes.fromByteArray(
+                                GetObjectResponse.builder().build(),
+                                mapper.writeValueAsString(account).getBytes())));
+
+        // when
+        final Future<Account> result = target.getAccountById("accountId", timeout);
+
+        // then
+        result.onComplete(context.succeeding(returnedAccount -> {
+            assertThat(returnedAccount.getId()).isEqualTo("accountId");
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void getAccountByIdShouldReturnTimeout(VertxTestContext context) {
+        // given
+        given(timeout.remaining()).willReturn(-1L);
+
+        // when
+        final Future<Account> result = target.getAccountById("account", timeout);
+
+        // then
+        result.onComplete(context.failing(cause -> {
+            assertThat(cause)
+                    .isInstanceOf(TimeoutException.class)
+                    .hasMessage("Timeout has been exceeded");
+
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void getAccountByIdShouldReturnAccountNotFound(VertxTestContext context) {
+        // given
+        given(s3AsyncClient.getObject(any(GetObjectRequest.class), any(AsyncResponseTransformer.class)))
+                .willReturn(CompletableFuture.failedFuture(
+                        NoSuchKeyException.create(
+                                "The specified key does not exist.",
+                                new IllegalStateException("error"))));
+
+        // when
+        final Future<Account> result = target.getAccountById("notFoundId", timeout);
+
+        // then
+        result.onComplete(context.failing(cause -> {
+            assertThat(cause)
+                    .isInstanceOf(PreBidException.class)
+                    .hasMessage("Account with id notFoundId not found");
+
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void getAccountByIdShouldReturnInvalidJson(VertxTestContext context) {
+        // given
+        given(s3AsyncClient.getObject(any(GetObjectRequest.class), any(AsyncResponseTransformer.class)))
+                .willReturn(CompletableFuture.completedFuture(
+                        ResponseBytes.fromByteArray(
+                                GetObjectResponse.builder().build(),
+                                "invalidJson".getBytes())));
+
+        // when
+        final Future<Account> result = target.getAccountById("invalidJsonId", timeout);
+
+        // then
+
+        result.onComplete(context.failing(cause -> {
```

Comment:
Will the test timeout or give a proper error message if it wouldn't fail?

Reply:
Yes.